### DISCUSSION TITLE
don't want to write each field test . I want overall test .please help me

### DISCUSSION THREAD

**Discussion by NarenderSiri** [react-hook-form/react-hook-form#9225] [General] [answered]
const schema = yup.object().shape({
    tax: yup.string().required("Tax is Required").test("","Remove white spaces" ,(value)=>!!value.trim()),
    taxName: yup.string().required("Tax Name is Required").test("","Remove white spaces" ,(value)=>!!value.trim()),
    countryName: yup.string().required("Country Name is Required").test("","Remove white spaces" ,(value)=>!!value.trim()),
  });

**Comment by bluebill1049** [comment]
Don't think there is alternative, schema will need to declare each input validation rules.

**Comment by Moshyfawn** [comment]
You can simply create a validation schema base / factory and extend it with field specific operators like:
```ts
const createBaseFieldSchema = (name: string) =>
  yup.string().required(`${name} is Required`).test("", "Remove white spaces", (value) => !!value.trim())

const schema = yup.object({
  tax: createBaseFieldSchema("tax").min(2, "2 MIN")
})
```

**Reply by NarenderSiri** [reply]
But name is getting undefined .
Example: undefined is required .

but I want tax is required

**Reply by Moshyfawn** [reply]
You should pass your field name to the `createBaseFieldSchema` to have a proper error message. See [this codesandbox](https://codesandbox.io/s/objective-wu-8py3in?file=/src/App.tsx)

**Comment by NarenderSiri** [comment, marked as answer]
It is not working.please check   
[codesandbox](https://codesandbox.io/s/confident-booth-45b4k6?file=/src/App.js)

**Reply by Moshyfawn** [reply]
Your factory function expects a named argument `name`, but you're passing a `string` instead. 
- Either update your function from `({ name })` to `(name)`
- or pass your field name to the factory function as a named argument `createBaseFieldSchema({ name: "tax })`

**Reply by NarenderSiri** [reply]
Tq so much

**Reply by Moshyfawn** [reply]
No problem!
